tests/components/tmdb/test_sensor.py

```diff
@@ -0,0 +1,107 @@
+"""The tests for The Movie Database platform."""
+import unittest
+from unittest.mock import patch
+
+from homeassistant.components.tmdb.sensor import LIST_TYPES, DOMAIN
+from homeassistant.setup import setup_component
+
+from tests.common import (get_test_home_assistant,
+                          MockDependency)
+
+import logging
+_LOGGER = logging.getLogger(__name__)
+
+MOVIE_CONFIG = {
+    'sensor': {
+        'platform': DOMAIN,
+        'api_key': 'test',
+        'movie_lists': LIST_TYPES,
+    }
+}
+
+TV_CONFIG = {
+    'sensor': {
+        'platform': DOMAIN,
+        'api_key': 'test',
+        'television_lists': LIST_TYPES,
+    }
+}
+
+MOCK_RESULTS = {
+    'results': [{}, {}]
+}
+
+MOCK_RESULTS_LENGTH = len(MOCK_RESULTS['results'])
+
+
+class MockTmdbSimple():
+    """Mock class for tmdbsimple library."""
+
+    def __init__(self):
+        """Add mock data for API return."""
+        self._data = MOCK_RESULTS
+
+    def upcoming(self):
+        """Return list upcoming movies."""
+        return self._data
+
+    def now_playing(self):
+        """Return list of now playing movies."""
+        return self._data
+
+    def popular(self):
+        """Return list popular movies."""
+        return self._data
+
+    def top_rated(self):
+        """Return list of top rated movies."""
+        return self._data
+
+
+class TestTmdbSetup(unittest.TestCase):
```

Comment:
Please use standalone pytest test function for tests. See deconz test suite for example.

Reply:
I need the setup and tear down hooks.

Reply:
Use pytest fixtures instead. They are awesome!